src/Domain/DonationService.php

```diff
@@ -846,26 +861,20 @@ public function attemptFundingAllocation(Donation $donation): void
     {
         // @todo-MAT-388: remove runWithPossibleRetry if we determine its not useful and unwrap body of function below
         $this->runWithPossibleRetry(
-            function () use ($donation) {
-                try {
-                    $this->donationRepository->allocateMatchFunds($donation);
-                } catch (\Throwable $t) {
-                    // warning indicates that we *may* retry, as it depends on whether this is in the last retry or
-                    // not.
-                    $this->logger->warning(sprintf('Allocation got error, may retry: %s', $t->getMessage()));
-
-                    $this->matchingAdapter->releaseNewlyAllocatedFunds();
+            retryable: fn (Donation $donation) => $this->donationRepository->allocateMatchFunds($donation),
+            actionName: 'allocate match funds',
+            donation: $donation,
+            onErrorTidy: function (Donation $donation) {
+                $this->logger->info(sprintf('Resetting after match error for UUID %s', $donation->getUuid()));
```

Comment:
I think it's probably clearer if we don't pass donation to runWithPossibleRetry when all that will do is pass it to the closures - and then it has to be documented as only taking closures that accept donations. Instead we can capture the donation to the closures before passing them. The short closure will auto-capture, the long one can use `use`.

```suggestion
            retryable: fn () => $this->donationRepository->allocateMatchFunds($donation),
            actionName: 'allocate match funds',
            onErrorTidy: function () use ($donation) {
                $this->logger->info(sprintf('Resetting after match error for UUID %s', $donation->getUuid()));
```

Reply:
Yes, I tried this and got confused when unit tests had issues. It looks like the new issue there was actually from me moving `$retryable` to the short syntax, and relying on the automatic capture you mentioned. Not sure why the tests weren't happy that way. I was getting e.g.
> ArgumentCountError: Too few arguments to function MatchBot\Domain\DonationService::MatchBot\Domain\{closure}(), 0 passed in /var/www/html/src/Domain/DonationService.php on line 239 and exactly 1 expecte

Reversed that change for now and removed the extraneous new arg, in 17b3e482df1a99f5d3ccee324ea0134b7457f5bc

Reply:
Thanks. Not sure what the issue with the short syntax was, tests pass for me when I make this edit:

```diff
         $this->runWithPossibleRetry(
-            retryable: function () use ($donation) {
-                $this->donationRepository->allocateMatchFunds($donation);
-            },
+            retryable: fn() => $this->donationRepository->allocateMatchFunds($donation),
             actionName: 'allocate match funds',
```